src/main/python/deep_qa/layers/encoders/shareable_gru.py

```diff
@@ -24,5 +24,9 @@ def call(self, x, mask=None, **kwargs):
             result_shape = K.int_shape(res)
             if input_shape[1] is not None and result_shape[1] is None:
                 shape = [d if d is not None else -1 for d in input_shape]
+                if self.output_dim != input_shape[-1]:
```

Comment:
i think we need the check on line 25 still because result shape isn't guaranteed to be 3 dimensional (`return_sequence=False`)